mongodbatlas/table_mongodbatlas_cluster.go

```diff
@@ -0,0 +1,272 @@
+				Type:        proto.ColumnType_JSON,
+			},
+			{
+				Name:        "replication_factor",
+				Description: "Number of replica set members. Each member keeps a copy of your databases, providing high availability and data redundancy.",
+				Type:        proto.ColumnType_INT,
+			},
+			{
+				Name:        "replication_spec",
+				Description: "Configuration of each region in the cluster. Each element in this object represents a region where Atlas deploys your cluster.",
+				Type:        proto.ColumnType_JSON,
+			},
+			{
+				Name:        "replication_specs",
+				Description: "Configuration for each zone in a Global Cluster. Each object in this array represents a zone where Atlas deploys nodes for your Global Cluster.",
+				Type:        proto.ColumnType_JSON,
+			},
+			{
+				Name:        "srv_address",
+				Description: "Connection string for connecting to the Atlas cluster. The +srv modifier forces the connection to use TLS. The mongoURI parameter lists additional options.",
+				Type:        proto.ColumnType_STRING,
+			},
+			{
+				Name: "state_name",
+				Description: `Condition in which the API resource finds the cluster when you called the resource. The resource returns one of the following states:
+
+		IDLE
+		CREATING
+		UPDATING
+		DELETING
+		DELETED
+		REPAIRING`,
+				Type: proto.ColumnType_STRING,
+			},
+			{
+				Name:        "connection_strings",
+				Description: "Set of connection strings that your applications use to connect to this cluster.",
+				Type:        proto.ColumnType_JSON,
+			},
+			{
+				Name: "version_release_system",
+				Description: `Release cadence that Atlas uses for this cluster. Atlas supports:
+
+	CONTINUOUS: Atlas automatically updates your cluster to the latest major and rapid MongoDB releases as they become available.
+	LTS: Atlas automatically updates your cluster to subsequent patch releases of this MongoDB version. Atlas doesn't update your cluster to newer rapid or major MongoDB releases as they become available.`,
+				Type: proto.ColumnType_STRING,
+			},
+
+			// Steampipe standard columns
+			{
+				Name:        "title",
+				Description: "Title of the resource.",
+				Type:        proto.ColumnType_STRING,
+				Transform:   transform.FromField("Name"),
+			},
+		},
+	}
+}
+
+func listAtlasClusters(ctx context.Context, d *plugin.QueryData, h *plugin.HydrateData) (interface{}, error) {
+	// Create client
+	config := GetConfig(d.Connection)
+	client, err := getMongodbAtlasClient(ctx, d)
+	if err != nil {
+		plugin.Logger(ctx).Error("mongodbatlas_cluster.listAtlasClusters", "connection_error", err)
+		return nil, err
+	}
+	// Retrieve the list of incidents
+	itemsPerPage := int64(500)
+	// Reduce the basic request limit down if the user has only requested a small number of rows
+	if d.QueryContext.Limit != nil && *d.QueryContext.Limit < itemsPerPage {
```

Comment:
What happens if someone sets a limit of 0? Does the API request throw an error?

Reply:
nope

Reply:
What response does the API give?